### PR TITLE
Add missing API functions and doxygen style documentation for C bindings

### PR DESCRIPTION
**Pull request by BenjaminRueth**
this PR resulted from splitting up https://github.com/precice/precice/pull/398

### CODE REVIEW

**Comment by IshaanDesai** [comment]
After a discussion with @BenjaminRueth I think it is a good idea to update the documentation of the C Bindings along with adding the missing API functions

**Comment by BenjaminRueth** [comment]
> I thought we remove `src/precice/bindings/c/Constant*` already. I am confused. We should keep it consistent to the C++ API.

In the current version on develop this is also the case. We have to merge develop into this branch before we can merge back.

**Comment by BenjaminRueth** [comment]
I merged develop into this branch. I just realized that `precicec_actionWriteInitialData` was missing in the C bindings and added this action. This was already the case when we removed the constants in https://github.com/precice/precice/commit/945eed22727b1fdafe9051c92684a7c6752e26bb#diff-5cd53866994afe4e9816951e94a0077f.

@uekerman do you know whether this action is missing intentionally or was it just forgotten?

File: src/precice/bindings/c/SolverInterfaceC.cpp [hunk 1]
```diff
@@ -104,6 +122,13 @@ void precicec_fulfilledAction ( const char* action )
   impl->fulfilledAction ( std::string(action) );
 }
 
+int precicec_hasMesh ( const char* meshName)
+{
+  assertion ( impl != nullptr );
+  std::string stringMeshName (meshName);
+  return impl->hasMesh (stringMeshName);
```
[the remark this answers is not in the document]
Comment (IshaanDesai) [reply src/precice/bindings/c/SolverInterfaceC.cpp]: Yes you are right, resolved in the forthcoming commit.

**Comment by uekerman** [comment]
@BenjaminRueth Was probably never needed. Nothing to worry about.

**Comment by BenjaminRueth** [comment]
@uekerman Is everything fine now from your point of view?

**Pull request merged**